sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -2000,4 +2000,15 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
       assert(setOfPath.size() == pathSizeToDeleteOnExit)
     }
   }
+
+  test("SPARK-21912 Creating ORC datasource table should check invalid column names") {
+    withTable("orc1") {
+      Seq(" ", "?", ",", ";", "{", "}", "(", ")", "\n", "\t", "=").foreach { name =>
+        val m = intercept[AnalysisException] {
+          sql(s"CREATE TABLE orc1 USING ORC AS SELECT 1 `column$name`")
```

Comment:
This is CTAS. How about `CREATE TABLE`?

Reply:
Yep.

Reply:
It seems to be the same situation with Parquet. `CREATE TABLE` passes but `SELECT` raises exceptions.
```scala
scala> sql("CREATE TABLE parquet1(`a b` int) using parquet")
res1: org.apache.spark.sql.DataFrame = []

scala> sql("select * from parquet1").show
org.apache.spark.sql.AnalysisException: Attribute name "a b" contains invalid character(s) among " ,;{}()\n\t=". Please use alias to rename it.;
```

Reply:
Do we need to add Datasource specific operation on `createDataSourceTables` for Parquet and ORC?

Reply:
@gatorsmile . I tried the following. We can add a check for `ParquetFileFormat`, but not for `OrcFileFormat`. Should I change the PR title and scope instead?

```scala
    table.provider.get.toLowerCase match {
      case "parquet" =>
        dataSource.schema.map(_.name).foreach(
          org.apache.spark.sql.execution.datasources.parquet.ParquetSchemaConverter.checkFieldName)
      case "orc" =>
        dataSource.schema.map(_.name).foreach(
          org.apache.spark.sql.hive.OrcRelation.checkFieldName)
    }
```

Reply:
I'll try in another way.